src/test/java/org/opensearch/integTest/ToolIntegrationTest.java

```diff
@@ -0,0 +1,226 @@
+                + "\",\n"
+                + "   \"headers\": {\n"
+                + "    \"content-type\": \"application/json\",\n"
+                + "    \"x-amz-content-sha256\": \"required\"\n"
+                + "   },\n"
+                + "   \"request_body\": \"{\\\"prompt\\\":\\\"${parameters.prompt}\\\", \\\"max_tokens_to_sample\\\":${parameters.max_tokens_to_sample}, \\\"temperature\\\":${parameters.temperature},  \\\"anthropic_version\\\":\\\"${parameters.anthropic_version}\\\" }\"\n"
+                + "  }\n"
+                + " ]\n"
+                + "}"
+        );
+    }
+
+    private void clusterSettings(boolean clean) throws IOException {
+        if (!clean) {
+            updateClusterSettings("plugins.ml_commons.only_run_on_ml_node", false);
+            updateClusterSettings("plugins.ml_commons.memory_feature_enabled", true);
+            updateClusterSettings("plugins.ml_commons.trusted_connector_endpoints_regex", List.of("^.*$"));
+        } else {
+            updateClusterSettings("plugins.ml_commons.only_run_on_ml_node", null);
+            updateClusterSettings("plugins.ml_commons.memory_feature_enabled", null);
+            updateClusterSettings("plugins.ml_commons.trusted_connector_endpoints_regex", null);
+        }
+    }
+
+    private String setupModelGroup() throws IOException {
+        Request request = new Request("POST", "/_plugins/_ml/model_groups/_register");
+        request
+            .setJsonEntity(
+                "{\n"
+                    + "    \"name\": \"test_model_group_bedrock-"
+                    + UUID.randomUUID()
+                    + "\",\n"
+                    + "    \"description\": \"This is a public model group\"\n"
+                    + "}"
+            );
+        Response response = executeRequest(request);
+
+        String resp = readResponse(response);
+
+        return JsonParser.parseString(resp).getAsJsonObject().get("model_group_id").getAsString();
+    }
+
+    private String setupLLMModel(String connectorId, String modelGroupId) throws IOException {
+        Request request = new Request("POST", "/_plugins/_ml/models/_register?deploy=true");
+        request
+            .setJsonEntity(
+                "{\n"
+                    + "    \"name\": \"Bedrock Claude V2 model\",\n"
+                    + "    \"function_name\": \"remote\",\n"
+                    + "    \"model_group_id\": \""
+                    + modelGroupId
+                    + "\",\n"
+                    + "    \"description\": \"test model\",\n"
+                    + "    \"connector_id\": \""
+                    + connectorId
+                    + "\"\n"
+                    + "}"
+            );
+        Response response = executeRequest(request);
+
+        String resp = readResponse(response);
+
+        return JsonParser.parseString(resp).getAsJsonObject().get("model_id").getAsString();
+    }
+
+    private String setupConversationalAgent(String modelId) throws IOException {
```

Comment:
I think a flow agent is enough to cover tool IT test like https://github.com/opensearch-project/skills/blob/34ae75f81ed47ecc2e4006d32343e12fa388678b/src/test/java/org/opensearch/integTest/NeuralSparseSearchToolIT.java#L94

Reply:
you are right, conversational agent is used to test mock LLM server. To run integration test for tools, flow agent is enough.